addons/mail/static/src/core/common/settings_model.js

```diff
@@ -45,31 +33,14 @@ export class Settings extends Record {
         },
     });
     messageSound = fields.Attr(true, {
-        compute() {
-            return browser.localStorage.getItem(MESSAGE_SOUND) !== "false";
-        },
-        /** @this {import("models").Settings} */
-        onUpdate() {
-            if (this.messageSound) {
-                browser.localStorage.removeItem(MESSAGE_SOUND);
-            } else {
-                browser.localStorage.setItem(MESSAGE_SOUND, "false");
-            }
-        },
+        localStorageKey: MESSAGE_SOUND,
+        localStorageGet: (ls) => ls.get() !== "false",
+        localStorageSet: (val, ls) => (val ? ls.remove() : ls.set("false")),
```

Comment:
This is really heavy and it's repeated multiple times across the PR. It's also unclear what it does. Could we simplify somehow?